crates/trie/sparse/src/state.rs

```diff
@@ -843,10 +845,10 @@ struct DecodedProofNodes {
     new_nodes: usize,
 }
 
-/// Decodes the proof nodes returning additional information about the number of total, skipped, and
-/// new nodes.
-fn decode_proof_nodes(
-    proof_nodes: ProofNodes,
+/// Does not decode the proof nodes, but instead just skips the nodes that are already revealed and
+/// returns additional information about the number of total, skipped, and new nodes.
+fn skip_revealed_nodes(
```

Comment:
IMO the mention of not decoding can be removed, as it can be confusing when reading this code without the context that previously we were decoding the nodes here

```suggestion
/// Filters the decoded nodes that are already revealed and returns additional information about the number of total, skipped, and new nodes.
fn filter_revealed_nodes(
```